homeassistant/components/sunsynkweb/config_flow.py

```diff
@@ -0,0 +1,61 @@
+"""Config flow for Sunsynk Inverter Web integration."""
+
+from __future__ import annotations
+
+import logging
+import traceback
+from typing import Any
+
+import aiohttp
+from pysunsynkweb.const import BASE_API
+from pysunsynkweb.exceptions import AuthenticationFailed
+from pysunsynkweb.session import SunsynkwebSession
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow as BaseConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class SunsynkConfigFlow(BaseConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Sunsynk Inverter Web."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        session = async_get_clientsession(self.hass)
+        if user_input is not None:
+            swebsession = SunsynkwebSession(
+                session, user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+            )
```

Comment:
```suggestion
        if user_input is not None:
            swebsession = SunsynkwebSession(
                async_get_clientsession(self.hass), user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
            )
```
it's only used once, and we should only add it if we need it